db_lesson.sql

```diff
@@ -0,0 +1,467 @@
+|        16 | 奥村あき       | okumura@gizumo.jp    |             2 |   25 |      1 | 2025-10-04 10:14:55 | 2025-10-04 10:14:55 |
+|        17 | 黒川みどり     | kurokawa@gizumo.jp   |             2 |   36 |      2 | 2025-10-04 10:14:55 | 2025-10-04 10:14:55 |
++-----------+----------------+----------------------+---------------+------+--------+---------------------+---------------------+
+16 rows in set (0.000 sec)
+
+
+
+
+
+
+
+
+
+-- Q5------------------------------------------------------------------------------------------------
+-- 年齢の降順で男性の名前と年齢を取得
+
+MariaDB [db_lesson]> select name, age from people where gender = 1 order by age desc;
++--------------+------+
+| name         | age  |
++--------------+------+
+| 村上さとる   |   53 |
+| 福田だいすけ |   42 |
+| 斎藤ゆうき   |   38 |
+| 工藤まさき   |   34 |
+| 井上だいすけ |   27 |
+| 高橋ゆうき   |   25 |
+| 奥村あき     |   25 |
+| 鈴木たかし   |   20 |
++--------------+------+
+8 rows in set (0.002 sec)
+
+
+
+
+
+
+
+-- Q6------------------------------------------------------------------------------------------------
+-- テーブル・レコード・カラムという3つの単語を適切に使用して、下記のSQL文を日本語で説明
+
+SELECT `name`, `email`, `age` FROM `people` WHERE `department_id` = 1 ORDER BY `created_at`;
+
+`people`テーブルから、`department_id`が1のレコードにおいて`name`, `email`, `age`の三つのカラムを取得する。またそれらは`created_at`というレコードを昇順に並べたときの順番で並べる。
+
+
+
+
+-- Q7------------------------------------------------------------------------------------------------
+-- 20代の女性と40代の男性の名前一覧を取得
+
+MariaDB [db_lesson]> select name from people where age between 20 and 29 and gender = 2 or age between 40 and 49 and gender = 1;
++--------------+
+| name         |
++--------------+
+| 田中ゆうこ   |
+| 福田だいすけ |
+| 浅川ゆり     |
+| 月島あいか   |
++--------------+
+4 rows in set (0.001 sec)
+
+
+
+
+
+-- Q8------------------------------------------------------------------------------------------------
+-- 営業部に所属する人だけを年齢の昇順で取得
+
+MariaDB [db_lesson]> select name, age from people where department_id = 1 order by age asc;
++------------+------+
+| name       | age  |
++------------+------+
+| 鈴木たかし |   20 |
+| 月島あいか |   23 |
+| 高橋ゆうき |   25 |
+| 羽村みずき |   30 |
+| 工藤まさき |   34 |
++------------+------+
+5 rows in set (0.000 sec)
+
+MariaDB [db_lesson]>
+
+
+
+
+
+
+-- Q9------------------------------------------------------------------------------------------------
+-- 開発部に所属している女性の平均年齢を取得
+
+MariaDB [db_lesson]> select avg(age) as average_age from people where gender = 2 and department_id = 2 group by gender;
++-------------+
+| average_age |
++-------------+
+|     30.2500 |
++-------------+
+1 row in set (0.001 sec)
+
+
+
+
+
+
+
+-- Q10------------------------------------------------------------------------------------------------
+-- 名前と部署名とその人が提出した日報の内容を同時に取得
+
+MariaDB [db_lesson]> select p.name, d.name, r.content
+    -> from people p inner join departments d on d.department_id = p.department_id join reports r on p.person_id = r.person_id;
```

Comment:
[nits]
改行してみるとわかりやすいかと思い改行してみましたが、
現在、下記のようにinner join句とjoin句（どちらも同じ意味）が混在しているので、可読性あげる意味でどちらかに統一しておきましょう！

```suggestion
select
　　p.name, d.name, r.content
from
　　people p
inner join
　　departments d on d.department_id = p.department_id
join
　　reports r on p.person_id = r.person_id;
```

Reply:
両方inner joinで統一しました。

Reply:
改めてプルリクエストする必要はありますでしょうか？